### PR TITLE
[quality] fix tslint 'typedef' errors

### LINKED ISSUE #5843: [quality] fix 'typedef' explicit return types
**Description**

Fix [**`typedef`**](https://palantir.github.io/tslint/rules/typedef/) tslint errors present in the repository. 
There are currently [**1092**](https://gist.github.com/vince-fugnitto/d539c03aa3f3524a4be84d8ad34e8e6e) failed checks as of master.
Updating to include `return types` is part of the [coding guidelines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines#explicit-return-type)

**Information**

The following typedef checks have been enabled:
1. `call-signature`: requires type definition for call signatures

<div align='center'>

Screen Shot 2019-08-02 at 10 28 15 AM

</div>

2. `property-declaration`: requires type definition for property declarations.

<div align='center'>

Screen Shot 2019-08-02 at 10 28 30 AM

</div>

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #5843

- updated `typedef` tslint warning into an error
- fixed failed `typedef` checks with the rules `call-signature` and `property-declaration`.

#### How to test
... 

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by akosyakov** [comment]
sorry @vince-fugnitto please rebase

**Comment by vince-fugnitto** [comment]
> sorry @vince-fugnitto please rebase

That's alright :) it was to be expected anyways!

**Comment by akosyakov** [comment]
@vince-fugnitto ping me when it is rebased

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto ping me when it is rebased

@akosyakov it's been rebased

File: packages/callhierarchy/src/browser/callhierarchy-context.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export class CallHierarchyContext implements Disposable {
         return symbols;
     }
 
+    // tslint:disable-next-line:typedef
```
Comment (akosyakov) [inline packages/callhierarchy/src/browser/callhierarchy-context.ts]: `monaco.editor.IReference<MonacoEditorModel>`  or there is some issue with it?

Comment (vince-fugnitto) [reply packages/callhierarchy/src/browser/callhierarchy-context.ts]: I tried previously and got the error:
`Namespace 'monaco.editor' has no exported member 'IReference'.`

```ts
 async getEditorModelReference(uri: string): Promise<monaco.editor.IReference<MonacoEditorModel>> { }
```

**Comment by vince-fugnitto** [comment]
I just updated the code to fix the instances in `examples/browser` and `examples/electron` since it was not covered by CI but was picked up when running `tslint -p . | grep "typedef"`

**Comment by vince-fugnitto** [comment]
@akosyakov @marcdumais-work are we fine to merge the PR?
It will likely break PRs attempting to be merged to `master` without rebasing first.

**Comment by akosyakov** [comment]
@vince-fugnitto I'm fine, just notify on dev channel on Spectrum and if something breaks pin corresponding devs or provide a fix PR.

**Pull request merged**